lib/datadog/appsec/extensions.rb

```diff
@@ -54,6 +54,12 @@ def ip_denylist=(arg)
           @settings.merge(dsl)
         end
 
+        def user_id_denylist=(arg)
+          dsl = AppSec::Configuration::DSL.new
+          dsl.user_id_denylist = arg
+          @settings.merge(dsl)
+        end
```

Comment:
We do not provide any kind of validation on the appsec settings. Is that intended? 

Thoughts @lloeki ?

Reply:
We kind of do, but [only for env vars](https://github.com/DataDog/dd-trace-rb/blob/65eabdbccace4e1de67aaf74032d902bd42dacf6/lib/datadog/appsec/configuration/settings.rb#L106-L114) because they need to be parsed from string to whatever value.

Note that converting (for env vars) is a different operation than validating setting types.

This is due for improvement but that will be part of the configuration revamp project (effort here would be wasted).